### PR TITLE
Feature/move me tab to new activity

### LINKED ISSUE #10495: Move the access to Me section from the Bottom Navigation Bar to a menu in My Site section
**Information Architecture Project** aims to:

> ...make it easier for users to complete their desired task by making our apps information architecture more obvious and its labeling more clear.

As part of this project we want to **move the access to the Me section**, from being a menu in the main Bottom Navigation Bar to being a dedicated icon in the options menu in the top right of the My Site section.

This task goes in parallel with the **move of the Create Post** from the Bottom Navigation Bar to a dedicated FAB in the My Site section (as of today 6th Sep 2019 under design). 

Once both these tasks are completed we will have a Bottom Navigation Bar no more with 5 but with 3 menu items (namely: My site, Reader and Notifications).

### PR DESCRIPTION
**Pull request by develric**
Fixes #10495 

As a part of the Information Architecture Project this PR aims to move the access to the Me section, from being a menu in the main Bottom Navigation Bar to being a dedicated icon in the My Site section options menu. (credits also to @planarvoid since great part of this code changes was done during a pair programming session).

The following images show a before and after of this:

 | Before | After |
|---|---|
| before | after |

**Notes**
Some notes on this current first iteration implementation:
- As explained in #10495 to get a "balanced" Bottom Navigation Bar with 3 items (My site, Reader, Me) this PR needs to be shipped together with the PR relevant to the post creation button to be transformed in a FAB (this is currently under design). For now we removed only the Me menu item from the nav bar
- There has been some elaboration if to use the user Gravatar or the generic user icon for the added menu. We used the "ic_user_circle_white_24dp" for this first iteration. We can decide if to stay with it or change to Gravatar in a future iteration. (is this correct @osullivanchris ?)
- To feature flag this enhancement, we used a boolean flag (`ME_ACTIVITY_AVAILABLE`) in the WordPress build.gradle [here](https://github.com/wordpress-mobile/WordPress-Android/blob/bf845ad88805ce2ab9059c6b986f4b2bf6382aef/WordPress/build.gradle#L86) . The feature is disabled for vanilla flavor and enabled in the other cases.

This is the current result:

<p align="center">
  
</p>

**Manual testing:**

1. Compile and run the wasabi flavor
2. Check that the Me menu item is not in the bottom nav bar
<p align="center">
  
</p>

3. Check that the me options menu icon
<p align="center">
  
</p>
 is only available in the My site section

4. Tap on each menu item in the bottom navbar to check that the correct screen is visualized
5. Tap on the me options menu icon to check that the Me section is visualized
6. Tap on the back arrow and check you go back to the My site section
7. Compile and run vanilla flavor
8. check that the My site Me option menu is not available
9. Check that you have all the 5 menu item in the Bottom nav bar
10. Smoke test to check that the menus activate the expected screens. 

**Update release notes:**

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt [hunk 1]
```diff
@@ -137,35 +147,39 @@ class WPMainNavigationView @JvmOverloads constructor(
...
+            R.id.nav_sites -> getPosition(MY_SITE)
+            R.id.nav_reader -> getPosition(READER)
+            R.id.nav_write -> getPosition(NEW_POST)
+            R.id.nav_me -> getPosition(ME)
+            else -> getPosition(NOTIFS)
         }
     }
 
+    private fun getPageForItemId(@IdRes itemId: Int): PageType {
+        return getPageType(getPositionForItemId(itemId))
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt]: I think I'd move the logic from `getPositionForItemId` to this method and call `getPosition(getPageForItemId(itemId))` there instead. What do you think? It feels a bit cleaner (and you don't have to call `getPosition` there multiple times.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt [hunk 2]
```diff
@@ -137,35 +147,39 @@ class WPMainNavigationView @JvmOverloads constructor(
...
+    }
+
     @IdRes
     private fun getItemIdForPosition(position: Int): Int {
-        return when (position) {
-            PAGE_MY_SITE -> R.id.nav_sites
-            PAGE_READER -> R.id.nav_reader
-            PAGE_NEW_POST -> R.id.nav_write
-            PAGE_ME -> R.id.nav_me
+        return when (pages().getOrNull(position)) {
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt]: What do you think about pulling this `pages().getOrNull(position)` into a separate method, something like `getPositionOrNull(position)` in the companion object?

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt (hunk 1 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt]: Definitely cleaner; changed here 5d14346b530edc53485582cc0225262fcb362428

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt (hunk 2 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt]: Makes sense 👍. Changes here 0708529161ebc4f065f0142d206f9a5a8d09bc3b

**Pull request merged**